application/src/main/java/com/github/pieter_groenendijk/model/Reservation.java

```diff
@@ -0,0 +1,56 @@
+package com.github.pieter_groenendijk.model;
+
+import jakarta.persistence.*;
+
+import java.util.Date;
+
+@Entity
+@Table(name = "reservation")
+public class Reservation {
+    @Id
+    @GeneratedValue(strategy = jakarta.persistence.GenerationType.IDENTITY)
+    private long reservationId;
+    @Column(name = "reservationDate", nullable = false)
+    @Temporal(TemporalType.DATE)
+    private Date reservationDate;
+    
+    // TODO: Check if this is correct. Just a reference to ProductId not Product entity
+    @Column(name = "productId", nullable = false)
+    private long productId;
+    @OneToOne
+    @JoinColumn(name = "membershipId", nullable = false, unique = true)
+    private Membership membershipId;
```

Comment:
Hadden we vastgesteld dat 1 abonnement altijd 1 reservatie kan doen? 

Naast dat zullen in de tabel ook oude reservaties staan. Op die manier zou je ook meerdere reservaties aan 1 abonnementen kunnen hebben.

Reply:
Ah ik interpreteerde dit als 1 reservering hoort bij 1 membership - fixed